pkg/vault/test/foundry/VaultLiquidityRate.t.sol

```diff
@@ -213,4 +223,67 @@ contract VaultLiquidityWithRatesTest is BaseVaultTest {
             bytes("")
         );
     }
+
+    function testRemoveLiquiditySingleTokenExactInWithRate__Fuzz(uint256 rateWstEth, uint256 rateDai) public {
+        rateWstEth = bound(rateWstEth, 1e16, 1e20);
+        rateDai = bound(rateDai, 1e16, 1e20);
+
+        RateProviderMock(address(rateProviders[wstethIdx])).mockRate(rateWstEth);
+        RateProviderMock(address(rateProviders[daiIdx])).mockRate(rateDai);
+
+        // Refresh lastBalancesLiveScaled18 with new rates.
+        vault.loadPoolDataUpdatingBalancesAndYieldFees(pool, Rounding.ROUND_DOWN);
+
+        // Fix pool liquidity, so there's enough liquidity to `removeLiquiditySingleTokenExactIn`. It adds enough
+        // liquidity so that both tokens have 50% of liquidity in the pool, when scaled18. Since both tokens have 18
+        // decimals, the only factor to consider is the rate of each one.
+        // Considering a pool with exactly poolInitAmount of each token, we need to add tokens in the side where the
+        // balance scaled18 is smaller. If rate of WstETH is 1, and DAI is 2, we have less WstETH than DAI, so we need
+        // to add more WstETH. The amount of WstETH to add is `(daiBalance * rateDai / rateWstEth - wstEthBalance)`.
+        // Since daiBalance and wstEthBalance is poolInitAmount, this formula becomes
+        // `poolInitAmount * (rateDai / rateWstEth - 1)`.
+        uint256[] memory amountsToAdd = new uint256[](2);
+        if (rateWstEth < rateDai) {
+            amountsToAdd[wstethIdx] = poolInitAmount * ((rateDai / rateWstEth) - 1);
+        } else {
+            amountsToAdd[daiIdx] = poolInitAmount * ((rateWstEth / rateDai) - 1);
```

Comment:
Hmm I think there's a mistake here.

The formula is correct, but in FP math this should probably be (e.g. for `if` clause):

```
poolInitAmount.mulDown(rateDai.divDown(rateWstEth) - 1e18)
```

There's no need to simplify further given than this is a test. I'd just use FP math with no shortcuts (it's a multiplication and a division after all so it can be done without FP math, but the `-1` will make it confusing.

Reply:
Doing the division first and subtracting a single wei is OK on paper, but you're losing a lot of precision here.

Reply:
Yeap, totally right.. I was multiplying and dividing first, which didn't lose precision. But I needed to remove poolInitAmount from the result, so I did the math simplification and wrote in the code as I thought... Anyway, I'll change it.